features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Endpoints/AWSLambda/Credentials.jsx

```diff
@@ -0,0 +1,214 @@
+
+const useStyles = makeStyles(theme => ({
+    typography: {
+        display: 'inline-block',
+    },
+    textField: {
+        marginLeft: theme.spacing(1),
+        marginRight: theme.spacing(1),
+        width: 300,
+    },
+    helpIcon: {
+        fontSize: 20,
+    },
+    banner: {
+        marginLeft: theme.spacing(1),
+        marginRight: theme.spacing(1),
+        marginTop: theme.spacing(1),
+        marginBottom: theme.spacing(1),
+    },
+}));
+
+/**
+ * The credentials component. This component holds the access and secret key pair.
+ * @param {any} props The parameters to credentials component.
+ * @returns {any} HTML view of the credentials component.
+ */
+export default function Credentials(props) {
+    const {
+        apiId,
+        endpointConfig,
+        endpointsDispatcher,
+        awsAccessMethod,
+        setAwsAccessMethod,
+    } = props;
+    const classes = useStyles();
+    const [pageError, setPageError] = useState(null);
+    const handleChange = (event) => {
+        const newEndpointConfig = { ...endpointConfig };
+        newEndpointConfig.amznAccessKey = '';
+        newEndpointConfig.amznSecretKey = '';
+        endpointsDispatcher({ action: 'set_awsCredentials', value: newEndpointConfig });
+        setAwsAccessMethod(event.target.value);
+        setPageError(null);
+    };
+    useEffect(() => {
+        API.getAmznResourceNames(apiId)
+            .then((response) => {
+                setPageError(response.body.error);
+            });
+    }, []);
+    return (
+        <React.Fragment>
+            <Typography className={classes.typography}>
+                <FormattedMessage
+                    id={'Apis.Details.Endpoints.EndpointOverview.awslambda' +
+                    '.endpoint.accessMethod'}
+                    defaultMessage='Access Method'
+                />
+            </Typography>
+            <RadioGroup aria-label='accessMethod' name='accessMethod' value={awsAccessMethod} onChange={handleChange}>
+                <div>
+                    <FormControlLabel
+                        value='role-supplied'
+                        control={<Radio color='primary' />}
+                        label={
+                            <FormattedMessage
+                                id={'Apis.Details.Endpoints.EndpointOverview.awslambda' +
+                                '.endpoint.accessMethod.roleSupplied'}
+                                defaultMessage='Using IAM role-supplied temporary AWS credentials'
+                            />
+                        }
+                    />
+                    <Tooltip
+                        title={
+                            <FormattedMessage
+                                id={'Apis.Details.Endpoints.EndpointOverview.awslambda' +
+                                '.endpoint.tooltip'}
+                                defaultMessage={'You can and should use an IAM role to manage temporary credentials ' +
```

Comment:
this message is not clear to me. Shall we reword?

Reply:
This is taken from aws docs. 
https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use_switch-role-ec2.html
(paragraph 2, line 1)